src/npctalk_funcs.cpp

```diff
@@ -741,6 +742,30 @@ void talk_function::buy_100_logs( npc &p )
     add_msg( m_good, _( "%s drops the logs off in the garage…" ), p.get_name() );
 }
 
+/*
+ * Function to make the npc drop non favorite, worn or wielded items at their current position.
+ */
+void talk_function::drop_items_in_place( npc &p )
+{
+    std::vector<drop_or_stash_item_info> to_drop;
+
+    // add all non favorite carried items to the drop off list
+    for( const item_location &npcs_item : p.all_items_loc() ) {
+        if( npcs_item.where() != item_location::type::character && !npcs_item->is_favorite ) {
```

Comment:
```suggestion
        if( !npcs_item->is_favorite && npcs_item.where() == item_location::type::container &&
            npcs_item.parent_item().where() == item_location::type::character ) {
```

I haven't tested it, but this should do the trick.

Reply:
Yeah it's working but it's a little confusing to me. Why is the type container working? I would have thought a container should only be items which can actually contain stuff? oO